src/libraries/System.Console/src/System/ConsolePal.iOS.cs

```diff
@@ -0,0 +1,189 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+
+namespace System
+{
+    internal sealed class NSLogStream : Stream
+    {
+        public override void Flush() { }
+
+        public override int Read(byte[] buffer, int offset, int count) => throw new NotSupportedException();
+
+        public override long Seek(long offset, SeekOrigin origin) => throw new NotSupportedException();
+
+        public override void SetLength(long value) => throw new NotSupportedException();
+
+        public override unsafe void Write(byte[] buffer, int offset, int count)
+        {
+            if (count > 0 && count <= buffer.Length - offset)
```

Comment:
Proper argument check (e.g. `ValidateWrite`) ?

Reply:
Oh, there is `ConsoleStream`, nice.